core/types.go

```diff
@@ -0,0 +1,18 @@
+package core
+
+import (
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/common"
+)
+
+// TODO: Unsure if this differs from the block defined with the header in common/types.go
+type Block struct {
```

Comment:
is this used somewhere? I see consensus/block type, do we need this as well

Reply:
Updates block structure to match spec better, and removed some unnecessary types. Lmk if you spot any immediate discrepancies, the spec seems a bit unclear IMO so we might need to clarify.

Reply:
I have updated the block structure to match the spec better. Lmk if you see any discrepancies

Reply:
looks good, see my other comment